src/Sentry.Unity/Il2CppEventProcessor.cs

```diff
@@ -17,12 +17,15 @@ internal class UnityIl2CppEventExceptionProcessor : ISentryEventExceptionProcess
     private static ISentryUnityInfo UnityInfo = null!; // private static will be initialized in the constructor
     private readonly Il2CppMethods _il2CppMethods;
 
-    public UnityIl2CppEventExceptionProcessor(SentryUnityOptions options, ISentryUnityInfo unityInfo)
+    public UnityIl2CppEventExceptionProcessor(SentryUnityOptions options)
     {
         Options = options;
-        UnityInfo = unityInfo;
-        _il2CppMethods = unityInfo.Il2CppMethods ?? throw new ArgumentNullException(nameof(unityInfo.Il2CppMethods),
-            "Unity IL2CPP methods are not available.");
+
+        // We're throwing here but this should never happen. We're validating UnityInfo before adding the processor.
+        UnityInfo = SentryPlatformServices.UnityInfo
```

Comment:
great! this was my only feedback in the other PR, validation that we're at the right state before going ahead

Also note that you copy a reference to use the same/consistent reference. This also means there's no path for a user to change these values after-the-fact.

But I understand that's the goal, and we then should follow this pattern when it makes sense. Instead of accessing the static mutable property in many places try to keep it to a minimum.